src/engine.rs

```diff
@@ -542,17 +563,20 @@ impl ApplyBackend for Engine {
 
```

Comment:
This won't work since `remove_all_storage` is empty right now.

Out of curiosity, when is the flat `reset_storage` set to true?

Reply:
Its a possible flag that can be passed back by the EVM. Technically, we don't wipe it, but I just have it in there for prosperity sake. Maybe there will eventually be a use for it. Lots of notes and discussion on that already. See engine.rs L#287